sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -774,6 +774,12 @@ object SQLConf {
       .doubleConf
       .createWithDefault(0.05)
 
+  val AUTO_UPDATE_SIZE =
+    buildConf("spark.sql.statistics.autoUpdate.size")
+      .doc("Enables automatic update for table size once table's data is changed.")
```

Comment:
This flag could slow down the whole data change commands. We need to clearly explain the potential performance regression. Based on the current description, all the users are willing to turn it on. Normally, the users expect we are doing the incremental updates, instead of recalculating it.